File: androidApp/v2/core/model/src/main/java/com/deadly/v2/core/model/Show.kt
package com.deadly.v2.core.model

import kotlinx.serialization.Serializable

/**
 * V2 Show domain model
 * 
 * Represents a Grateful Dead concert as a pure domain entity.
 * Contains show-level metadata and references to recordings,
 * but not the full recording objects themselves.
 */
@Serializable
data class Show(
    val id: String,
    val date: String,
    val year: Int,
    val band: String,
    val venue: Venue,
    val location: Location,
    val setlist: Setlist?,
    val lineup: Lineup?,
    
    // Recording references
    val recordingIds: List<String>,
    val bestRecordingId: String?,
    
    // Show-level stats (precomputed from recordings)
    val recordingCount: Int,
    val averageRating: Float?,
    val totalReviews: Int,
    
    // User state  
    val isInLibrary: Boolean,
    val libraryAddedAt: Long?
) {
    /**
     * Display title for the show
     */
    val displayTitle: String
        get() = "${venue.name} - $date"
    
    /**
     * Whether this show has ratings
     */
    val hasRating: Boolean
        get() = averageRating != null && averageRating > 0f
    
    /**
     * Formatted rating display
     */
    val displayRating: String
        get() = averageRating?.let { "%.1f★".format(it) } ?: "Not Rated"
    
    /**
     * Whether this show has multiple recordings
     */
    val hasMultipleRecordings: Boolean
        get() = recordingCount > 1
}

File: iosApp/deadly/Core/Model/Show.swift
import Foundation

struct Venue: Codable, Sendable, Equatable {
    let name: String
    let city: String?
    let state: String?
    let country: String

    var displayLocation: String {
        var parts: [String] = []
        if let city { parts.append(city) }
        if let state { parts.append(state) }
        if country != "USA" { parts.append(country) }
        return parts.joined(separator: ", ")
    }
}

struct Location: Codable, Sendable, Equatable {
    let displayText: String
    let city: String?
    let state: String?

    static func fromRaw(_ raw: String?, city: String?, state: String?) -> Location {
        let parts = [city, state].compactMap { $0 }
        let display = raw ?? (parts.isEmpty ? "Unknown Location" : parts.joined(separator: ", "))
        return Location(displayText: display, city: city, state: state)
    }
}

struct Show: Codable, Sendable, Equatable, Identifiable {
    let id: String
    let date: String
    let year: Int
    let band: String
    let venue: Venue
    let location: Location
    let setlist: Setlist?
    let lineup: Lineup?

    // Recording references
    let recordingIds: [String]
    let bestRecordingId: String?

    // Show-level stats (precomputed from recordings)
    let recordingCount: Int
    let averageRating: Float?
    let totalReviews: Int

    // Cover art
    let coverImageUrl: String?

    // User state
    let isInLibrary: Bool
    let libraryAddedAt: Int64?

    var displayTitle: String { "\(venue.name) - \(date)" }
    var hasRating: Bool { averageRating.map { $0 > 0 } ?? false }
    var displayRating: String { averageRating.map { String(format: "%.1f★", $0) } ?? "Not Rated" }
    var hasMultipleRecordings: Bool { recordingCount > 1 }
}
